amd/src/tribute.js

```diff
@@ -31,1148 +72,1218 @@ if (!Array.prototype.find) {
+            if (!this.isContentEditable(targetElement)) {
+              if (targetElement !== this.getDocument().activeElement) {
+                targetElement.focus();
+              }
+            } else {
+              this.selectElement(targetElement, path, offset);
+            }
           }
+        }, {
+          key: 'replaceTriggerText',
+          value: function replaceTriggerText(text, requireLeadingSpace, hasTrailingSpace) {
+            var context = this.tribute.current;
+            this.resetSelection(context.element, context.selectedPath, context.selectedOffset);
+
+            var info = this.getTriggerInfo(requireLeadingSpace, true, hasTrailingSpace);
+
+            // Create the event
+            var replaceEvent = new CustomEvent('tribute-replaced', {detail: text});
+
+            if (info !== undefined) {
+              if (!this.isContentEditable(context.element)) {
+                var myField = this.getDocument().activeElement;
+                text += ' ';
+                var startPos = info.mentionPosition;
+                var endPos = info.mentionPosition + info.mentionText.length + 1;
+                myField.value = myField.value.substring(0, startPos) + text + myField.value.substring(endPos, myField.value.length);
+                myField.selectionStart = startPos + text.length;
+                myField.selectionEnd = startPos + text.length;
+              } else {
+                // Add a space to the end of the pasted text
+                text += '\xA0';
+                this.pasteHtml(text, info.mentionPosition, info.mentionPosition + info.mentionText.length + 1);
+              }
 
-          return window.getSelection();
-        }
-      }, {
-        key: 'getNodePositionInParent',
-        value: function getNodePositionInParent(element) {
-          if (element.parentNode === null) {
-            return 0;
+              context.element.dispatchEvent(replaceEvent);
+            }
           }
-
-          for (var i = 0; i < element.parentNode.childNodes.length; i++) {
-            var node = element.parentNode.childNodes[i];
-
-            if (node === element) {
-              return i;
+        }, {
+          key: 'pasteHtml',
+          value: function pasteHtml(html, startPos, endPos) {
+
+            var range = void 0,
+                sel = void 0;
+            sel = this.getWindowSelection();
+            range = this.getDocument().createRange();
+            range.setStart(sel.anchorNode, startPos);
+            range.setEnd(sel.anchorNode, endPos);
+            range.deleteContents();
+
+            var el = this.getDocument().createElement('div');
+            el.innerHTML = html;
+            var frag = this.getDocument().createDocumentFragment(),
+                node = void 0,
+                lastNode = void (0);
+            while (node != el.firstChild) {
+              lastNode = frag.appendChild(el.firstChild);
+            }
```

Comment:
Change made for the selected item to be appended. Variable `node  = void 0 ` is  always not equal to el. firstChild